cmd/anonymizer/app/query/query_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package query
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/jaegertracing/jaeger/cmd/query/app"
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/plugin/metrics/disabled"
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
+	dependencyStoreMocks "github.com/jaegertracing/jaeger/storage/dependencystore/mocks"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	spanstoremocks "github.com/jaegertracing/jaeger/storage/spanstore/mocks"
+)
+
+var (
+	matchContext = mock.AnythingOfType("*context.valueCtx")
+	matchTraceID = mock.AnythingOfType("model.TraceID")
+
+	mockInvalidTraceID = "xyz"
+	mockTraceID        = model.NewTraceID(0, 123456)
+
+	mockTraceGRPC = &model.Trace{
+		Spans: []*model.Span{
+			{
+				TraceID: mockTraceID,
+				SpanID:  model.NewSpanID(1),
+				Process: &model.Process{},
+			},
+			{
+				TraceID: mockTraceID,
+				SpanID:  model.NewSpanID(2),
+				Process: &model.Process{},
+			},
+		},
+		Warnings: []string{},
+	}
+)
+
+type testServer struct {
+	address    net.Addr
+	server     *grpc.Server
+	spanReader *spanstoremocks.Reader
+}
+
+func newQuery(t *testing.T, addr string) *Query {
```

Comment:
I think it's better to use the New function from the package instead of repeating it. It also looks like the Query struct is missing a Close function, to close its internal connection.